File: 2022/01/part-two.js
const fs = require("fs");

const data = fs.readFileSync("./input", "utf8");
const array = data.split("\n");

let max = [];
let sum = 0;
for (let i = 0; i < array.length; i++) {
  const line = array[i];
  if (line == "") {
    if (sum > 0) {
      max.push(sum);
    }
    sum = 0;
  } else {
    sum = sum + Number(line);
  }
}
if (sum > 0) {
  max.push(sum);
}

max.sort((a, b) => a - b);
console.log(max[max.length - 1] + max[max.length - 2] + max[max.length - 3]);

File: 2022/01/part-two.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"sort"
	"strconv"
)

func readFile() int {
	inputFile, err := os.Open("./input")
	if err != nil {
		panic(err)
	}
	scanner := bufio.NewScanner(inputFile)

	max := make([]int, 0)
	sum := 0
	for scanner.Scan() {
		line := scanner.Text()
		if line == "" {
			if sum > 0 {
				max = append(max, sum)
			}
			sum = 0
		} else {
			number, err := strconv.Atoi(line)
			if err != nil {
				panic(err)
			}
			sum += number
		}

	}
	if sum > 0 {
		max = append(max, sum)
	}

	sort.Slice(max, func(i, j int) bool {
		return max[i] < max[j]
	})

	return max[len(max)-1] + max[len(max)-2] + max[len(max)-3]
}

func main() {
	max := readFile()
	fmt.Println(max)
}
